doc/adr/019_Automated_Content_Deployment.md

```diff
@@ -0,0 +1,147 @@
+# Automated Content Deployment
+
+Date: 2021-03-12
+
+Deciders: Abban Dunne, Corinna Hillebrand, Gabriel Birke
+
+Technical Story: https://phabricator.wikimedia.org/T262704
+
+## Status
+
+Accepted
+
+## Context and Problem Statement
+
+We want to automatically deploy the fundraising content repository
+whenever there is a change (i.e. a Git commit on the `test` or `production`
+branch) to the destination servers associated with the branch.
+
+
+## Decision Drivers
+
+* For security reasons, we want to host the system that does the
+  deployment on our own infrastructure.
+* The repository consists of several files. We want the deployment to be
+  *atomic*, i.e our application either uses the new set of files or the old
+  one, but never a mix.
+* We have three production servers that need to receive the new version
+  simultaneously (although not atomically).
+* We would like the system to notify us (via email) when a deployment failed.
+* We would like to have a record (e.g. log file) of successful deployments.
+* We have an Ansible playbook that does the atomic deployment to the test
+  and production servers.
+* We have a `Dockerfile` definition that packages Ansible, the deployment
+  playbooks and our server configuration into a standalone Docker image.
+  The deployment software can run this Docker image to do the deployment.
+* We prefer "configuration as code" over setting up workflows in a GUI.
+* The maintenance and onboarding effort for the system should be as low 
+  as possible.
+* (Optional) It would be nice if we had a "button" to trigger the
+  deployment
+* (Optional) It would be nice if branches of the application could be
```

Comment:
Yeah, that sentence doesn't make much sense. But I know what I wanted to say, so I'll rewrite it as "It would be nice if feature branches of the application repository could be associated with feature branches in the content repository (instead of using `test`)."

Does that make it clearer?

Reply:
Yep 👍